app/auth/forms.py

```diff
@@ -47,8 +48,14 @@ class RegistrationForm(FlaskForm):
     password = PasswordField('Password', validators=[
         DataRequired(), 
         EqualTo('password2', message='Passwords must match.'),
-        Length(8,64, message='Password must be at least 8 characters in length.'),
-        PasswordStrength()
+        Length(min=8, message='Password must be at least 8 characters in length.'),
+        PasswordStrength(),
+        Regexp(
+            '^(?=.*[0-9].*[0-9]).*$',
+            0, 'Passwords must contain at least two digits.'),
+        Regexp(
+            '^(?=.*[A-Z].*[A-Z]).*$',
+            0, 'Passwords must contain at least two uppercase letters.')
```

Comment:
Tbh II think this whole PR should probably go on hold until after the IA review, rather than trying to pre-empt what they will say!